packages/apollo/src/commands/service/check.ts

```diff
@@ -296,79 +305,129 @@ export default class ServiceCheck extends ProjectCommand {
                    * name: implementing service name inside of the graph
                    * sha: git commit hash/docker id. placeholder for now
                    */
-                  task.output = "Creating composed schema against the graph";
                   const {
                     errors,
                     warnings,
                     compositionValidationDetails
                   } = await project.engine.checkPartialSchema({
                     id: configName,
                     graphVariant: tag,
-                    implementingServiceName: flags.serviceName || info.name,
+                    implementingServiceName: serviceName,
                     partialSchema: {
                       sdl: info.sdl
                     }
                   });
 
                   // FIXME: reformat to match other check results
-
                   taskOutput.federation = {
                     errors,
                     warnings
                   };
 
                   if (compositionValidationDetails) {
                     schemaHash = compositionValidationDetails.schemaHash;
-                  } else {
-                    // FIXME: We should provide some better error handling at the GraphQL layer for this
-                    throw new Error(`Federated service could not be composed due to the following errors:
-  ${errors && errors.map(err => (err && err.message) || "").join("\n")}
-                  `);
                   }
-                } else {
-                  schema = await project.resolveSchema({ tag: config.tag });
 
-                  await gitInfo(this.log);
+                  if (errors.length > 0) {
+                    const decodedErrors = errors.filter(Boolean).map(error => {
```

Comment:
`isNotNullOrUndefined` from `apollo-env` solves this nicely for us at the type level 👍

Reply:
OMG why can `Boolean` _not_ do this 🤕